articles/linear-regression-code/index.md

```diff
@@ -0,0 +1,135 @@
+
+### Introduction
+
+As discussed in the previous article, linear regression is a type of supervised learning algorithm. It is used in many applications in the financial industry. First, let us understand the various functions needed to implement a linear regression class, to begin with the coding aspect. 
+
+### Code Structure
+
+We define the following methods in the class `Regressor`:
+1. `__init__`: In the `__init__` method, we initialize all the parameters with default values. These parameters are added as and when required. For now, you will see that all the parameters are initialized beforehand. But while coding, you create new variables as and when required.
+2. `initialize_weights_and_bias`: In the `initialize_weights_and_bias` method, the weights and biases are initialized. We use random initialization to initialize the weights, and the bias is initially 0. 
+3. `computeError`: This function calculates the error or loss function and returns the cost. The input to this function is the predicted output and the actual output. 
+4. `optimize`: This function uses stochastic gradient descent to optimize the loss function. We initially compute the gradients of the weights and the bias in the variables `dW` and `db.` Using these gradients, we updated our weights and biases iteratively.
+5. `normalize`: This function subtracts the mean from the data and divides it by its standard deviation. This ensures the data is centered around 0, and the standard deviation is always 1. Data with such distribution is easier to work with and results in the model learning better. This is because various features have various scales. The larger values may end up contributing more to the output. Hence, normalization ensures no such anomalies take place. 
+6. `fit`: The fit method calls all the above functions. In this method, we perform normalization on the input features and compute the loss. Once the loss is computed, we optimize the loss function by applying the `optimize` function on the input-output pair. We perform this until there is no significant change in the loss values obtained after training.
+7. `predict`: This function is used to test the model on unseen data. The input to the function is the input data. The predict function outputs the dependent variable.
+8. `plot`: In this method, we plot the loss function versus the number of iterations.
+9. `score`: This function calculates the accuracy. The accuracy is computed using the following formula:
+    $$ accuracy = \frac{(y - \hat{y})^2}{\sum{_i}^n(y-\bar{y})^2}$$
+
+### Code 
+
+Let us put together the information we collected above and create the `Regressor` class. 
+
+```py
+import numpy as np
```

Comment:
It would be a good idea to add comments to your code, thereby making it easier for a beginner to follow.